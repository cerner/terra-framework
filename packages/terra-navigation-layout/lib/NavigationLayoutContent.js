"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _reactRouterDom = require("react-router-dom");
var _bind = _interopRequireDefault(require("classnames/bind"));
var _RoutingStack = _interopRequireDefault(require("./RoutingStack"));
var _configurationPropTypes = require("./configurationPropTypes");
var _NavigationLayoutContentModule = _interopRequireDefault(require("./NavigationLayoutContent.module.scss"));
var _excluded = ["navigationLayoutRoutes", "redirectPath", "stackNavigationIsEnabled"];
var cx = _bind.default.bind(_NavigationLayoutContentModule.default);
var propTypes = {
  /**
   * The routing configuration from which Routes will be generated.
   */
  navigationLayoutRoutes: _configurationPropTypes.processedRoutesPropType,
  /**
   * The String path to redirect to should the routes specified by the routeConfig all fail to match.
   */
  redirectPath: _propTypes.default.string,
  /**
   * Flag to enable navigation within the RoutingStack.
   */
  stackNavigationIsEnabled: _propTypes.default.bool
};
var NavigationLayoutContent = function NavigationLayoutContent(_ref) {
  var navigationLayoutRoutes = _ref.navigationLayoutRoutes,
    redirectPath = _ref.redirectPath,
    stackNavigationIsEnabled = _ref.stackNavigationIsEnabled,
    customProps = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: cx('content')
  }, /*#__PURE__*/_react.default.createElement(_RoutingStack.default, {
    navEnabled: stackNavigationIsEnabled,
    routes: navigationLayoutRoutes,
    ancestorProps: customProps
  }, redirectPath && /*#__PURE__*/_react.default.createElement(_reactRouterDom.Redirect, {
    to: redirectPath
  })));
};
NavigationLayoutContent.propTypes = propTypes;
var _default = NavigationLayoutContent;
exports.default = _default;