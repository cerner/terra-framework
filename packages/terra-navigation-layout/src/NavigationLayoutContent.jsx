import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import classNames from 'classnames/bind';

import RoutingStack from './RoutingStack';
import { processedRoutesPropType } from './configurationPropTypes';

import styles from './NavigationLayoutContent.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The routing configuration from which Routes will be generated.
   */
  navigationLayoutRoutes: processedRoutesPropType,
  /**
   * The String path to redirect to should the routes specified by the routeConfig all fail to match.
   */
  redirectPath: PropTypes.string,
  /**
   * Flag to enable navigation within the RoutingStack.
   */
  stackNavigationIsEnabled: PropTypes.bool,
};

const NavigationLayoutContent = ({
  navigationLayoutRoutes, redirectPath, stackNavigationIsEnabled, ...customProps
}) => (
  <div className={cx('content')}>
    <RoutingStack
      navEnabled={stackNavigationIsEnabled}
      routes={navigationLayoutRoutes}
      ancestorProps={customProps}
    >
      {redirectPath && <Redirect to={redirectPath} />}
    </RoutingStack>
  </div>
);

NavigationLayoutContent.propTypes = propTypes;

export default NavigationLayoutContent;
