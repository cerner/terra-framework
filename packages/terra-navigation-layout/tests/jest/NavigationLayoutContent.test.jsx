import React from 'react';
import NavigationLayoutContent from '../../src/NavigationLayoutContent';

describe('NavigationLayoutContent', () => {
  // Snapshot Tests
  it('should render a default NavigationLayoutContent', () => {
    const wrapper = shallow(<NavigationLayoutContent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a NavigationLayoutContent with provided props', () => {
    const wrapper = shallow(<NavigationLayoutContent
      navigationLayoutRoutes={[]}
      redirectPath="/"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
