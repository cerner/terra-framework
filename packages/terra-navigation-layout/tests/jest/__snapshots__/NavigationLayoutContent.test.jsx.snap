// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLayoutContent should render a NavigationLayoutContent with provided props 1`] = `
<div
  className="content"
>
  <withRouter(RoutingStack)
    ancestorProps={Object {}}
    routes={Array []}
  >
    <Redirect
      to="/"
    />
  </withRouter(RoutingStack)>
</div>
`;

exports[`NavigationLayoutContent should render a default NavigationLayoutContent 1`] = `
<div
  className="content"
>
  <withRouter(RoutingStack)
    ancestorProps={Object {}}
  />
</div>
`;
